### PR TITLE
[Feature 17] CompleteTaskUC and BuyItemUC

### PR DESCRIPTION
**Pull request by XinyingXu**
# Description

Modify CompleteTask, and documentation and tests.

Fixes #60 #62 

## Type of change

- [x] Compilation error fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation
- [x] Test cases
- [ ] Redundant code removal (fix code smells)
- [ ] Data storage 
- [ ] UI element

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/test/java/useCases/CompleteTaskUCTest.java [hunk 1]
```diff
@@ -1,11 +1,48 @@
...
+     * toDoList should have one task
+     * doneList should have one task
+     * task2 is completed after deadline so the user should have 0 point
+     * Then test CompleteTaskUC.completeTask() by completing task1
+     * task1 is completed before deadline so the user should have 3 points
+     * toDoList should have zero task
+     * doneList should have two tasks
+     */
+    @Test
+    public void CompleteTaskTest() throws AbsentTaskNameException {
```
Comment (harshsbajwa) [inline src/test/java/useCases/CompleteTaskUCTest.java:27]: This test is too general, please split into multiple cases

File: src/test/java/useCases/BuyItemUCTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * The remaining points should be 1
+ * Customization should have equipped item
+ * Equips Straw Hat
+ * Second Test: Try to buy the hat "Cap" with price too high
+ * The purchase should be unsuccessful
+ * Third Test: Try to buy the hat "Baseball Cap" which is already unlocked
+ * The remaining points should be 1
+ * Equips Baseball Cap
+ */
+public class BuyItemUCTest {
```
Comment (harshsbajwa) [inline src/test/java/useCases/BuyItemUCTest.java:22]: This test is too general, please split into multiple tests

**Review by harshsbajwa — CHANGES_REQUESTED** [review]
Please split tests

File: src/main/java/entities/Customization.java [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
     // the current item equipped by the pet
     public Item current_equipment;
 
-    
+
     // ==CONSTRUCTORS==
 
     public Customization(){
         this.current_equipment = null;
         this.is_currently_equipped = false;
```
Comment (harshsbajwa) [inline src/main/java/entities/Customization.java:24]: change to camelCase

**Review by stephaniesaab — CHANGES_REQUESTED** [review]
I've left some comments regarding extendability and testing. These should be quick changes to implement.

**Review by stephaniesaab — APPROVED** [review]
Great job implementing the changes suggested.

**Review by serenahonekin — APPROVED** [review]
Both use cases look good in documentation and the implementation of a user uc

**Review by harshsbajwa — APPROVED** [review]
good work

**Pull request merged**